### ISSUE TITLE
Using multiple roles

### ISSUE THREAD

**Issue by olivier3400** [LeaseWeb/LswSecureControllerBundle#4]
When you specifiy several roles une the @secure annotation an AND is done beetwen the roles.
It is more logical to do an OR as done int the JMS bundle because you can filter it in the controller function.

To do it you must change in the Lsw\SecureControllerBundle\Security\ControllerListener onKernelController function

```
            if (!$this->securityContext->isGranted($role)) {
                throw new AccessDeniedException(
                    'Current user is not granted required role "'.$role.'".'
                );
            }
        }
```

by

```
            if ($this->securityContext->isGranted($role)) return;
        }
        throw new AccessDeniedException('Current user is not granted');
```

**Comment by Gadgetdude** [comment]
Another option that will give you a lot of flexibility is using the `@Security` annotation provided by the `SensioFrameworkExtraBundle` (Symfony 2.4+ feature). Use the expression language to filter through multiple roles, set up custom conditions, or anything you like. If you need something really complicated you should probably be looking into another solution than annotations (perhaps a guard (Symfony 2.8+ feature) or a simple listener/subscriber).

```
/**
 * @Security("has_role('ROLE_USER') or has_role('ROLE_ASSISTANT')")
 */
```
Source: http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/security.html

**Comment by mevdschee** [comment]
@Gadgetdude I agree and I added a note to the readme pointing people in that direction.
